src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/profiler/ProfilerEditingTarget.java

```diff
@@ -671,16 +696,61 @@ public void execute()
       };
    }
    
-   private static void onMessage(String message, String file, int line)
-   {  
+   private void onMessage(final String message,
+                          final String file,
+                          final String details,
+                          final int line)
+   {
       if (message == "sourcefile")
       {
-         FilePosition filePosition = FilePosition.create(line, 0);
-         CodeNavigationTarget navigationTarget = new CodeNavigationTarget(file, filePosition);
-         
-         RStudioGinjector.INSTANCE.getFileTypeRegistry().editFile(
-               FileSystemItem.createFile(navigationTarget.getFile()),
-               filePosition);
+         server_.profileSources(file, new ServerRequestCallback<String>()
+         {
+            @Override
+            public void onResponseReceived(String response)
+            {
+               selectedPath_ = file;
+               selectedLine_ = line;
+               hasValidPath_ = !StringUtil.isNullOrEmpty(response);
+               
+               commands_.gotoProfileSource().setEnabled(hasValidPath_);
+               
+               if (details == "open")
+               {
+                  if (hasValidPath_)
+                  {
+                     FilePosition filePosition = FilePosition.create(line, 0);
+                     CodeNavigationTarget navigationTarget = new CodeNavigationTarget(file, filePosition);
+                     
+                     fileTypeRegistry_.editFile(
+                           FileSystemItem.createFile(navigationTarget.getFile()),
+                           filePosition);
+                  }
+                  else
+                  {
+                     globalDisplay_.showMessage(GlobalDisplay.MSG_ERROR,
+                           "Error while opening profiler source",
+                           "Could not find reference to: " + selectedPath_);
```

Comment:
Wording nit: this is a missing file rather than a missing file reference. Maybe "The source file selectedPath_ does not exist."?